src/include/ikarus/finiteElements/mechanics/linearElastic.hh

```diff
@@ -170,6 +171,27 @@ namespace Ikarus {
       }
     }
 
+    void calculateAt(const ResultRequirementsType& req, const Eigen::Vector<double, Traits::mydim>& local,
+                     ResultTypeMap<double>& result) const {
+      using namespace Dune::Indices;
+      using namespace Dune::DerivativeDirections;
+      using namespace Dune;
+
+      const auto eps = getStrainFunction(req.getFERequirements());
+      const auto C   = getMaterialTangent();
+      auto gp        = toDune(local);
+      auto epsVoigt  = eps.evaluate(gp, on(gridElement));
+
+      auto cauchyStress = C * epsVoigt;
```

Comment:
without eval this only works since `C` and `epsVoigt` are in the same scope. with eval the return type is here a vector. without it is some rpoxy object that stores this matrix vector product operation. The actual conversion takes place at `resultVector = cauchyStress;`, the operator=() will force the evaluation of the right-hand side, since the lefthand side is a real EIgen::Matrix object and no some expression template.

If `C` and `epsVoigt`  would be not in the same scope of  `resultVector = cauchyStress;` the evaluation would use some invalidated memory and we would produce undefined behaviour.

Her for performance reasons its nice, since the evaluation takes one place, if the if statement later is true.
Therefore, in the end we can keep it the way it is or we move the stress calc inside.

Reply:
Thanks for the detailed comment. We can then change it to `auto cauchyStress = (C * epsVoigt).eval();`